File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/cpp/cwe-476/code_before1.cpp
// From cwe-snippets, snippets_90/non-compliant/C++/cwe-0476/char_33.cpp

void func()
{
    char * data;
    char * &dataRef = data;
    data = NULL;
    {
        char * data = dataRef;
        printHexCharLine(data[0]);
    }
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/cwe-94/code_before1.java
// From cwe-snippets, snippets_2/non-compliant/Java/0290.java   

VelocityContext context = new VelocityContext();
context.put( "name", user.name );
// Load the template
String template = getUserTemplateFromRequestBody(request);
RuntimeServices runtimeServices = RuntimeSingleton.getRuntimeServices();
StringReader reader = new StringReader(template);
SimpleNode node = runtimeServices.parse(reader, "myTemplate");
template = new Template();
template.setRuntimeServices(runtimeServices);
template.setData(node);
template.initDocument();
// Render the template with the context data
StringWriter sw = new StringWriter();
template.merge( context, sw );
